### ISSUE TITLE
I get this error when i try to send a large file.

### ISSUE THREAD

**Issue by ttheekshanaw** [andyet/SimpleWebRTC#519]

**Comment by fippo** [comment]
how large is the file? The remote browser might have crashed...

**Comment by ttheekshanaw** [comment]
Well the file size i tried was 200mb. It randomly stops around 30%. The browser did not crash as well.

**Comment by ttheekshanaw** [comment]
when i try the same file opening two tabs on the same computer it works perfectly. But If i do it with 2 different computers i get that error.

**Comment by ttheekshanaw** [comment]
File size: if file size is reasonably small and can be stored and loaded as one Blob, you can load into memory using the File API and then send the file over a reliable channel as is (though bear in mind that browsers impose limits on maximum transfer size). As file size get larger, things get trickier. A chunking mechanism is required: file chunks are loaded and sent to another peer, accompanied with chunkId metadata so the peer can recognize them. Note that in this case you will also need to save the chunks first to offline storage (for example, using the FileSystem API) and only when you have the file in its entirety save it to the user's disk.

source - https://www.html5rocks.com/en/tutorials/webrtc/datachannels/

**Comment by fippo** [comment]
SimpleWebRTC implements a chunking mechanism but buffers in memory at the receiver. The filesystem API didn't make it to the browser.

**Comment by ttheekshanaw** [comment]
Any solution for this problem ?

**Comment by ttheekshanaw** [comment]
fippo are you sure its a chrome bug. I feel like its not

**Comment by fippo** [comment]
did your peerconnection ever get established? see [example 1 here](http://testrtc.com/webrtc-api-trace/) for how to find out.

**Comment by ttheekshanaw** [comment]
yes it did. It Just drops in the middle of the transfer. As I said earlier If i am testing the same 200MB file from same laptop with different tabs it works perfectly. But if i use different laptops on the same network it gives that error.

**Comment by ttheekshanaw** [comment]
Hello,
   I think it is a synchronization issue. For some reason the sending rate is higher that the receiving speed. If i make sending slower intentionally it works. I was able to send a large file by slowing the sending rate.

**Comment by fippo** [comment]
I could reproduce it on some networks. That is definetly a regression in chrome, i'll see if I can get enough native logs to file a bug :-/

**Comment by fippo** [comment]
sorry for the delay. I filed https://bugs.chromium.org/p/webrtc/issues/detail?id=7051

It might help to merge https://github.com/otalk/filetransfer/pull/7/ which changes how backpressure is handled but i'd like to hear an opinion from the chrome folks first.

**Comment by fippo** [comment]
can you try again please?
